### PR TITLE
Add support for umask when exec container

### PR DESCRIPTION
**Pull request by Wang-squirrel**
Signed-off-by: Wang-squirrel <wang.xiaosong1@zte.com.cn>

### CODE REVIEW

**Comment by ningmingxiao** [comment]
this will support update umask when exec container  @kolyshkin

File: libcontainer/setns_init_linux.go [hunk 1]
```diff
@@ -59,6 +59,10 @@ func (l *linuxSetnsInit) Init() error {
 			return err
 		}
 	}
+	 if l.config.Config.Umask != nil {
+               unix.Umask(int(*l.config.Config.Umask))
+       }
+
```
Comment (AkihiroSuda) [inline libcontainer/setns_init_linux.go:65]: Could you add an integration test?

**Comment by Wang-squirrel** [comment]
@kolyshkin done

**Comment by Wang-squirrel** [comment]
Please check it, thanks @kolyshkin  @AkihiroSuda

**Comment by AkihiroSuda** [comment]
shfmt is failing
https://github.com/opencontainers/runc/actions/runs/3545124237/jobs/5984445102

**Comment by Wang-squirrel** [comment]
> shfmt is failing https://github.com/opencontainers/runc/actions/runs/3545124237/jobs/5984445102

done it

**Review by kolyshkin — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]
LGTM but slightly fear this may potentially break some compatibility

**Comment by kolyshkin** [comment]
> LGTM but slightly fear this may potentially break some compatibility

I guess we can afford it with 1.2.0. I checked and it looks like crun is supporting umask for exec from the very beginning.

**Comment by kolyshkin** [comment]
@Wang-squirrel can you please rebase?

**Comment by Wang-squirrel** [comment]
> @Wang-squirrel can you please rebase?

Done

**Review by kolyshkin — APPROVED** [review]
still LGTM

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**